components/hab/src/cli/hab/util.rs

```diff
@@ -18,29 +27,99 @@ use structopt::StructOpt;
 use url::Url;
 
 #[derive(ConfigOpt, StructOpt)]
+#[configopt(derive(Serialize))]
 #[structopt(no_version)]
 #[allow(dead_code)]
 pub struct AuthToken {
-    /// Authentication token for Builder
+    /// Authentication token for Builder.
     #[structopt(name = "AUTH_TOKEN", short = "z", long = "auth")]
-    auth_token: Option<String>,
+    pub value: Option<String>,
 }
 
 #[derive(ConfigOpt, StructOpt, Deserialize)]
 #[configopt(derive(Serialize))]
 #[structopt(no_version)]
 #[allow(dead_code)]
 pub struct BldrUrl {
-    /// Specify an alternate Builder endpoint. If not specified, the value will be taken from
-    /// the HAB_BLDR_URL environment variable if defined. (default: https://bldr.habitat.sh)
-    // TODO (DM): This should probably use `env` and `default_value`
-    #[structopt(name = "BLDR_URL", short = "u", long = "url")]
-    bldr_url: Option<Url>,
+    /// Specify an alternate Builder endpoint. If not specified, the value will be
+    /// taken from the HAB_BLDR_URL environment variable if defined. (default: https://bldr.habitat.sh)
+    #[structopt(name = "BLDR_URL",
+                short = "u",
+                long = "url",
+                validator = valid_url)]
+    pub value: Option<Url>,
+}
+
+#[derive(ConfigOpt, StructOpt, Deserialize, Serialize)]
+#[structopt(no_version)]
+#[configopt(derive(Serialize))]
+#[allow(dead_code)]
+pub struct BldrOrigin {
+    /// The Builder origin name to target
+    #[structopt(name = "ORIGIN", short = "o", long = "origin", validator = valid_origin)]
+    pub name: Origin,
+}
+
+pub fn bldr_url_from_env_load_or_default() -> String {
+    bldr_url_from_env().unwrap_or_else(|| {
+                           match config::load() {
+                               Ok(config) => {
+                                   match config.bldr_url {
+                                       Some(v) => v,
+                                       None => DEFAULT_BLDR_URL.to_string(),
+                                   }
+                               }
+                               Err(e) => {
+                                   println!("Found a cli.toml but unable to load it. Resorting \
```

Comment:
Maybe use `error!` instead of `println!`

Reply:
Modified as such in https://github.com/habitat-sh/habitat/pull/7841/commits/633e75036f3864b3ac7d2cfbf02a8a4704ce4120